### PR TITLE
Move fzy algo from rff crate into local file and some other small tweaks to Rust code

### PR DESCRIPTION
**Pull request by ImmemorConsultrixContrarie**
Original (MIT licensed) "rff" crate has `terminal` module which utilizes `std::os::unix` thus it doesn't compile on non-unix OS, and could not be "cargo check"ed and tested.
If it is not accepted, all tweaks that don't depend on moving fzy algorithm are done in "Less allocations, more laziness" commit.

### CODE REVIEW

File: pythonx/clap/extracted_fzy/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/target
+!.cargo
+*.lock
```
Comment (liuchengxu) [inline pythonx/clap/extracted_fzy/.gitignore]: I don't think the [faq](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries) applies to our situation. Although it's a library, it's unpublished and actually used for vim-clap exclusively. No others depend on it, and we need to ensure people are always able to compile this Rust extension on their own. Keeping `Cargo.lock` is better for our case.

**Comment by liuchengxu** [comment]
Overall looks good to me, except ignoring the `Cargo.lock`  I'm not against moving the fzy algo impl into vim-clap if that helps non-unix users. We can use a dedicated fzy library once https://github.com/stewart/rff/issues/2 is resolved, but rff is seemingly not much active now :(.

**Comment by ImmemorConsultrixContrarie** [comment]
Okay, will revert gitignoring .lock files and add my .locks to the PR, once I'm back at home.

**Comment by ImmemorConsultrixContrarie** [comment]
Ready to merge now, I guess?
@liuchengxu

**Comment by liuchengxu** [comment]
Merged, thanks!

**Pull request merged**